src/coreclr/jit/lower.cpp

```diff
@@ -6793,6 +6793,40 @@ void Lowering::LowerStoreIndirCommon(GenTreeStoreInd* ind)
     if (!comp->codeGen->gcInfo.gcIsWriteBarrierStoreIndNode(ind))
     {
         LowerStoreIndir(ind);
```

Comment:
The pattern in `lower.cpp` is that `LowerCommonX` methods call `LowerX` methods. It may be that `LowerStoreIndir` has modified `ind` by the time we check we check for our optimization opportunity.

So I suggest a reorder: move the optimization before `LowerStoreIndir`. You should then be able to delete the `#if` for `SetContained`, as it would be dead code - `LowerStoreIndir` should check for the containment as appropriate.

Also: is containing _all_ FP constants worth it on ARM/ARM64? The codegen for things like `*x = 1.1` should be checked.

Reply:
Thank you for your comments.




> So I suggest a reorder: move the optimization before `LowerStoreIndir`. You should then be able to delete the `#if` for `SetContained`, as it would be dead code - `LowerStoreIndir` should check for the containment as appropriate.

I'll make a call to LowerStoreInd after optimization.

> Also: is containing _all_ FP constants worth it (does it even work?) on ARM/ARM64? The codegen for things like `*x = 1.1` should be checked.

You made the right remarks. I will fix them and come back with new changes soon.